src/Database/Query.php

```diff
@@ -132,10 +133,10 @@ class Query implements ExpressionInterface, IteratorAggregate
     /**
      * Constructor.
      *
-     * @param \Cake\Datasource\ConnectionInterface $connection The connection
+     * @param \Cake\Database\ConnectionInterface $connection The connection
      * object to be used for transforming and executing this query
      */
-    public function __construct($connection)
+    public function __construct(ConnectionInterface $connection)
```

Comment:
It's a rather simple fix though, people would just need to change the `use` statement and possibly the doc block. However, given that the guidelines say that breaking changes should go into the next major version, introducing the least possible amount of BC incompatibilities is probably the way to go.